contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1190,6 +1202,26 @@ pltsql_bbfViewHasInsteadofTrigger(Relation view, CmdType event)
 	return false;
 }
 
+static Datum
+bbf_trunc_numeric_result(Node *fn_expr, Datum result, Oid result_type, int32 result_typmod)
+{
+	int32       scale;
+
+	if (result && OidIsValid(result_type) && getBaseType(result_type) == NUMERICOID)
+	{
+		if (fn_expr != NULL && result_typmod == -1)
+			result_typmod = tsql_get_numeric_typmod_from_exp(NULL, fn_expr);
+
+		if (result_typmod != -1)
+		{
+			scale = (result_typmod - VARHDRSZ) & 0xffff; 
+			return DirectFunctionCall2(numeric_trunc, result, Int32GetDatum(scale));
```

Comment:
Should we be calling tsql_numeric_trunc?

Reply:
`tsql_numeric_trunc` is a wrapper function which internally calls `numeric_trunc` when the third parameter passed to function `tsql_numeric_trunc` is non-zero, else it will call `tsql_numeric_round`. But in our case we only need to truncate the result hence there is no benefit of using `tsql_numeric_trunc`.